### PR TITLE
Fix the command `ipilot-agent istio-clean-iptables` cannot clean up the iptables rules generated for istio dns proxy

### LINKED ISSUE #47957: The command `pilot-agent istio-clean-iptables` cannot clean up the iptables rules generated for istio dns proxy
### Is this the right place to submit this?

- [X] This is not a security vulnerability or a crashing bug
- [X] This is not a question about how to use Istio

### Bug Description

In scenarios where virtual machines are used as Istio workloads, it is often necessary to change the relevant parameters of the `pilot-agent` subcommand to ensure that they meet the expectations of the production environment. In order to describe the problem simply and intuitively, I extract the key information and describe it as follows:

According to the startup logic of the official script and combined with the configuration parameters in my environment, before starting `pilot-agent proxy`, I need to use `pilot-agent istio-iptables --proxy-gid 980 --proxy-uid 983 --capture-all -dns true --redirect-dns true` to write iptables rules in the environment.
Corresponding to the above, after I stop `pilot-agent proxy`, I need to use `pilot-agent istio-clean-iptables --proxy-gid 980 --proxy-uid 983 --capture-all-dns true --redirect-dns true` to clean up the iptables rules written in the environment.

Unfortunately, `pilot-agent istio-clean-iptables` currently does not support the `--capture-all-dns` parameter. This directly leads to the fact that after executing the `pilot-agent istio-clean-iptables` command, there are still iptables rules related to the istio dns proxy remaining. As a user, I have to use the following two commands to clean up:
```
$ iptables -t nat -F
$ iptables -t raw -F
```

### Version

```Text
$ istioctl version
client version: 1.17.8
control plane version: 1.17.8
data plane version: 1.17.8 (11 proxies)

$ kubectl version --short
Client Version: v1.24.4
Server Version: v1.24.4
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by singhwang**
Fix #47957

I added the `--capture-all-dns` parameter to `pilot-agent istio-clean-iptables`. In the parameter description and usage, make it consistent with the `--capture-all-dns` parameter of `pilot-agent istio-iptables`.

### CODE REVIEW

**Review by hzxuzhonghu — COMMENTED** [review]
LGTM

/ok-to-test

File: releasenotes/notes/47957.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+issue: 
+  - 47957
+releaseNotes:
+  - |
+    **Fixed** the command ipilot-agent istio-clean-iptables cannot clean up the iptables rules generated for istio dns proxy.
```
Comment (GregHanson) [inline releasenotes/notes/47957.yaml:8]:
```suggestion
    **Fixed** an issue where the `pilot-agent istio-clean-iptables` command cannot clean up the iptables rules generated for istio dns proxy.
```

Comment (singhwang) [reply releasenotes/notes/47957.yaml]: ok, I'll adjust it, thanks for your help.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**